### PR TITLE
fix(build): watcher of webpack watches too many files including output files and files in node_modules

### LINKED ISSUE #3672: [DX problem] webpack keeps compiling repeatedly without file changes
Start it from my terminal without any editor open.

image

~~I guess its' because the `tsc` process is running, and it compiles `ts` into `js`, then triggers webpack to recompile.~~

### PR DESCRIPTION
**Pull request by UncleBill**
closes #3672

### CODE REVIEW

File: packages/maskbook/webpack.config.ts [hunk 1]
```diff
@@ -48,6 +48,9 @@ function EnvironmentPluginNoCache(def: Record<string, any>) {
     }
     return new DefinePlugin(next)
 }
+const watchOptions = {
+    ignored: /\bnode_modules\b/,
```
Comment (Jack-Works) [inline packages/maskbook/webpack.config.ts:52]: So... Who is changing node_modules?

**Comment by UncleBill** [comment]
> So... Who is changing node_modules?

@Jack-Works  Maybe us.

Let's checkout the comparison of with and without ignoring:

| without ignoring | with ignoring |
|---|---|
| image | image |

We can see the watcher without ignoring watches `packages/maskbook/node_modules`, and it watches 49776 files!

**Comment by Jack-Works** [comment]
Ok so please also remove `WatchMissingModulesPlugin` because I guess it does not work with node_modules not watched.

**Comment by UncleBill** [comment]
> Ok so please also remove WatchMissingModulesPlugin because I guess it does not work with node_modules not watched.

Done.

**Review by Jack-Works — APPROVED** [review]

**Review by Jack-Works — APPROVED** [review]

**Review by nuanyang233 — APPROVED** [review]

**Pull request merged**